File: learntoprogram/PackageDemoB.java
/**
* Description: 包技术
* 不同包间的访问,继承的子类可访问protected的权限
* @author fnfly2005
*/
package mypack.subpack;
public class PackageDemoB extends mypack.PackageDemoA
{
    protected void run() //父类protected只对子类可见，而对子类的实例不可见,需要覆盖后使用
    {
        super.run();
    }
}

File: learntoprogram/pypack/PackageDemoB.py
#!/usr/bin/python
#coding:utf-8
"""
Description: 继承导入模块
"""
import PackageDemoA as pa
class PackageDemoB(pa.PackageDemoA):
    def run(self):
        print "PackageDemoB run!!"

if __name__ == '__main__':
    p = PackageDemoB()
    p.show()
    p.run()
